libs/iterator_support/include/hpx/iterator_support/traits/is_sentinel_for.hpp

```diff
@@ -0,0 +1,59 @@
+//  Copyright (c) 2020 Giannis Gonidelis
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once
+
+#include <hpx/iterator_support/traits/is_iterator.hpp>
+#include <hpx/type_support/always_void.hpp>
+
+#include <utility>
+
+namespace hpx { namespace traits {
+
+    template <typename Sent, typename Iter, typename Enable = void>
+    struct is_sentinel_for : std::false_type
+    {
+    };
+
+    template <typename Sent, typename Iter>
+    struct is_sentinel_for<Sent, Iter,
+        typename util::always_void<typename is_iterator<Iter>::type,
```

Comment:
Got it! That should have been `typename std::enable_if<is_iterator<Iter>::value>::type, `. That may cause the problems. Must have missed it on some rebase.